RockLib.HealthChecks.WebApi/HealthCheckHandlerExtensions.cs

```diff
@@ -2,55 +2,62 @@
+    /// <param name="route">The route of the health endpoint.</param>
+    /// <param name="indent">Whether to indent the JSON output.</param>
+    public static void MapHealthRoute(this HttpRouteCollection routes,
+        string? healthCheckRunnerName = null, string route = "/health", bool indent = false)
+        => routes.MapHealthRoute(HealthCheck.GetRunner(healthCheckRunnerName)!, route, indent);
+
+
+    /// <summary>
+    /// Adds a health check route to the route collection.
+    /// </summary>
+    /// <param name="routes">The route collection.</param>
+    /// <param name="healthCheckRunner">
+    /// The <see cref="IHealthCheckRunner"/> to use. If <see langword="null"/>,
+    /// the value of the <see cref="HealthCheck.GetRunner"/> method is used.
+    /// </param>
+    /// <param name="route">The route of the health endpoint.</param>
+    /// <param name="indent">Whether to indent the JSON output.</param>
+    public static void MapHealthRoute(this HttpRouteCollection routes,
+        IHealthCheckRunner healthCheckRunner, string route = "/health", bool indent = false)
     {
-        private static int _routeIndex = -1;
-
-        /// <summary>
-        /// Adds a health check route to the route collection.
-        /// </summary>
-        /// <param name="routes">The route collection.</param>
-        /// <param name = "healthCheckRunnerName" >
-        /// The name of the <see cref="IHealthCheckRunner"/> to use. If <see langword="null"/> or not provided,
-        /// the default value of the <see cref="HealthCheck.GetRunner"/> method is used.
-        /// </param>
-        /// <param name="route">The route of the health endpoint.</param>
-        /// <param name="indent">Whether to indent the JSON output.</param>
-        public static void MapHealthRoute(this HttpRouteCollection routes,
-            string healthCheckRunnerName = null, string route = "/health", bool indent = false)
-            => routes.MapHealthRoute(HealthCheck.GetRunner(healthCheckRunnerName), route, indent);
-        
-
-        /// <summary>
-        /// Adds a health check route to the route collection.
-        /// </summary>
-        /// <param name="routes">The route collection.</param>
-        /// <param name="healthCheckRunner">
-        /// The <see cref="IHealthCheckRunner"/> to use. If <see langword="null"/>,
-        /// the value of the <see cref="HealthCheck.GetRunner"/> method is used.
-        /// </param>
-        /// <param name="route">The route of the health endpoint.</param>
-        /// <param name="indent">Whether to indent the JSON output.</param>
-        public static void MapHealthRoute(this HttpRouteCollection routes,
-            IHealthCheckRunner healthCheckRunner, string route = "/health", bool indent = false)
+#if NET6_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(routes);
+#else
+        if (routes is null) { throw new ArgumentNullException(nameof(routes)); }
+#endif
+
+        if (string.IsNullOrWhiteSpace(route))
```

Comment:
why aren't you using the NET6_0_OR_GREATER conditional and ThrowIfNull here as well?